astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -48,10 +48,13 @@ def test_testwarn(tw):
 
 
 class TestUfuncHelpers:
-    # Note that this test should work even if scipy is present, since
-    # the scipy.special ufuncs are only loaded on demand.
+    # Note that this test may fail even if scipy is present, since
+    # the scipy.special ufuncs are only loaded on demand. This is because
```

Comment:
Why "even"? Now this is assumed to pass if `not HAS_SCIPY`, right?
As I understood the logic wouldn't that rather be
```suggestion
    # Note that this test may fail if scipy is present, although the
    # scipy.special ufuncs are only loaded on demand. This is because
```
The `reason` to me also sounds more like "scipy coverage is not included" – there simply is no list `all_scipy_ufuncs` to be compared against, whether such a list would be incomplete or not?
Lastly, as the test apparently _may_ pass, perhaps it should rather be marked `xfail`? Though that may not be very informative...

Reply:
We have `xfail_strict` on, so if it only sometimes fail, you should not use `pytest.mark.xfail` because it will fail if it passes. 🤪

Reply:
In that context, if I do re-enable the call to `scipy.special.wofz(z)` (as a Quantity) in `Voigt1D`, skipping `test_coverage` does not prevent `test_scipy_registered` and `test_scipy_registration` from failing (which still expect that no scipy ufuncs have been loaded before). So those tests still depend on the prior state in an unfortunate way.

It seems `test_coverage` could actually be fixed in a slightly longer way by adding
```python
        if HAS_SCIPY:
            from scipy import special as sps

            all_sps_ufuncs = {
                ufunc
                for ufunc in sps.__dict__.values()
                if isinstance(ufunc, np.ufunc)
            }
        else:
            all_sps_ufuncs = set()

        assert all_q_ufuncs - all_np_ufuncs - all_erfa_ufuncs - all_sps_ufuncs == set()
```
But that would still not help with the scipy_registered test, and of course skipping them is pointless.

Reply:
Your suggestion is definitely better than just skipping!

p.s. `scipy_registered` is independent of whether any quantity has been put through a scipy ufunc (registration is independent of actually having `scipy`).

Reply:
So, where is this discussion going? I see the status as "approved" but looks like @dhomeier still has concern?